providers/tentacle.rb

```diff
@@ -172,6 +172,16 @@
     end
     new_resource.updated_by_last_action(actions_updated?([register_polling_instance]))
   else
+
+    firewall = windows_firewall_rule 'Octopus Deploy Tentacle' do
+      localport port.to_s
+      dir :in
+      protocol 'TCP'
+      firewall_action :allow
+      action :create
+      only_if { node['octopus']['tentacle']['instance']['manage_firewall'] }
```

Comment:
I would make this a resource value called `manage_firewall` or `configure_firewall`.  I like configure more I think.

```ruby 
manage_firewall = new_resource.manage_firewall
```